### ISSUE TITLE
Error: "No Available Profile" when "uploadOnSave" is set to true in .vscode/sftp.json

### ISSUE THREAD

**Issue by evtimov** [satiromarra/vscode-sftp#26]
Thank you for this great extension!

If you don't mind, I would like to report a small bug (using Windows 10) which prevents "uploadOnSave": true in .vscode/sftp.json to work as expected.

CTRL+S saves the file but it doesn't actually upload it.

Instead the extension produces following error: 

`No Available Profile.`
`[error] Error: Config Not Found.`

This is my .vscode/sftp.json file:
Remote explorer works just fine, I am able to manually browse, upload and download files.

```
{
    "name": "My Project",
    "host": "192.168.1.140",
    "protocol": "ftp",
    "port": 21,
    "username": "ftpusers",
    "password": "ftppassword",
    "remotePath": "public_html/",
    "uploadOnSave": true
}
```

Please find attached a screenshot that shows that message. 
There is nothing in the Output/Debug Console unfortunately.

![2023-04-05_202448](https://user-images.githubusercontent.com/47464444/230185716-a80740e8-f830-47c0-8a4a-4cdfe856713d.png)

**Comment by satiromarra** [comment]
@evtimov Is it still failing you?

**Comment by evtimov** [comment]
Hi satiromarra, yes:

```
Version: 1.77.1 (user setup)
Commit: b7886d7461186a5eac768481578c1d7ca80e2d21
Date: 2023-04-04T23:21:11.906Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Windows_NT x64 10.0.19045
Sandboxed: Yes
```

`SFTP Extension version: v1.17.10`

```
{
    "name": "Project 1",
    "host": "example-project-domain.com",
    "username": "ftpuser",
    "password": "ftppassword",
    "remotePath": "public_html/",
    "ignore": [
        "tmp/",
        "temp/",
        ".vscode/",
        ".git/",
        "_notes/",
        ".well-known/",
        ".DS_Store"
    ],
    "useTempFile": false, "openSsh": false, 
    "protocol": "ftp", "secure": false, "port": 21, "showHidden": true,
    "passive": true,
    "uploadOnSave": true
}
```

![111](https://user-images.githubusercontent.com/47464444/230889711-97764e19-1d04-43e6-9e34-1c5f62a819dd.png)

**Remote explorer works just fine, I am able to manually browse, upload and download files.**

**Comment by satiromarra** [comment]
@evtimov Can you put the following configuration in the file `sftp.json`? `"context":"."`

**Comment by evtimov** [comment]
@satiromarra added `"context":"."` in `sftp.json` but still the same... BUT everything works just fine on my laptop with same config and software versions - both VS Code and the SFTP Extension. There must be something wrong on my Desktop Workstation not related to VS Code at all. I suppose you may close this issue now. Thank you very much once again.

**Comment by evtimov** [comment]
I have managed to replicate this issue @satiromarra 

I use a small Windows tool called "**Visual Subst**", which is basically a tool that allows you to associate a local directory with a virtual drive (drive letter). I suppose similar if not the same as mapped network drives. 

To summarize, working from a **_local folder_** `e:\\work\\project1` is fine but if I start working from a **_local virtual drive_** `v:\\work\\project1\`  then the issue appears as described above.

Can you think of a fix please?

**Comment by satiromarra** [comment]
@evtimov I'm not sure if I found the problem, can you try this [beta](https://github.com/satiromarra/vscode-sftp/releases/download/v1.17.15-beta1/code-sftp-1.17.15-beta1.vsix)? 

I don't know what implications it will have for the rest of users to remove the part that is giving you problems, that is why it is better to create an exception for these cases.

In your `sftp.json` add the option `"ignoreRealPath":false`

```
{
    "name": "Project 1",
    "host": "example-project-domain.com",
    "username": "ftpuser",
    "password": "ftppassword",
    "remotePath": "public_html/",
    "ignore": [
        "tmp/",
        "temp/",
        ".vscode/",
        ".git/",
        "_notes/",
        ".well-known/",
        ".DS_Store"
    ],
    "useTempFile": false, "openSsh": false, 
    "protocol": "ftp",
    "secure": false,
    "port": 21,
    "showHidden": true,
    "passive": true,
    "uploadOnSave": true,
    "ignoreRealPath": false
}
```

**Comment by evtimov** [comment]
@satiromarra I confirm that adding `"ignoreRealPath":` **`true`** to `sftp.json` solves the issue.

¡Gracias por todo vuestro esfuerzo!

**Comment by satiromarra** [comment]
Published v1.17.16 🎉
